frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.tsx

```diff
@@ -0,0 +1,89 @@
+import { t } from "ttag";
+import { useEffect, useMemo, useState } from "react";
+import Modal from "metabase/components/Modal";
+import { SearchFilterModalFooter } from "metabase/search/components/SearchFilterModal/SearchFilterModalFooter";
+import {
+  FilterTypeKeys,
+  SearchFilterComponent,
+  SearchFilterKeys,
+  SearchFilters,
+} from "metabase/search/types";
+import Button from "metabase/core/components/Button";
+import { Title, Flex } from "metabase/ui";
+import { TypeFilter } from "./filters/TypeFilter";
+import { SearchFilterWrapper } from "./SearchFilterModal.styled";
+
+const filterMap: Record<FilterTypeKeys, SearchFilterComponent> = {
+  [SearchFilterKeys.Type]: TypeFilter,
+};
+
+export const SearchFilterModal = ({
+  isOpen,
+  setIsOpen,
+  value,
+  onChangeFilters,
+}: {
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+  value: SearchFilters;
+  onChangeFilters: (filters: SearchFilters) => void;
+}) => {
+  const [output, setOutput] = useState<SearchFilters>(value);
+
+  useEffect(() => {
+    setOutput(value);
+  }, [value]);
+
+  const closeModal = () => {
+    setIsOpen(false);
+  };
+
+  const clearFilters = () => {
+    onChangeFilters({});
+    setIsOpen(false);
+  };
+
+  const applyFilters = () => {
+    onChangeFilters(output);
+    setIsOpen(false);
+  };
+
+  // we can use this field to control which filters are available
+  // - we can enable the 'verified' filter here
+  const availableFilters: FilterTypeKeys[] = useMemo(() => {
+    return [SearchFilterKeys.Type];
```

Comment:
Correct me if I'm wrong, but are we using the type as a value here? i.e. is this returning
```js
["type"]
```

https://github.com/metabase/metabase/blob/6c9345abceb9e8f9e7d69cb86ec86c922e53b248/frontend/src/metabase/search/types.ts#L17-L19